File: Src/Icm.ConsoleMvc/TypeExtensions.vb
Imports System.Linq
Imports System.Runtime.CompilerServices

Public Module TypeExtensions
		<Extension> _
		Function GetAttributes(Of T As Attribute)(ByVal type As Type, ByVal inherit As Boolean) As T()
				Return DirectCast(type.GetCustomAttributes(GetType(T), inherit), T())
		End Function
		<Extension> _
		Function HasAttribute(Of T As Attribute)(ByVal type As Type, ByVal inherit As Boolean) As Boolean
				Return DirectCast(type.GetCustomAttributes(GetType(T), inherit), T()).Count > 0
		End Function
		<Extension> _
		Function GetAttribute(Of T As Attribute)(ByVal type As Type, ByVal inherit As Boolean) As T
				Return DirectCast(type.GetCustomAttributes(GetType(T), inherit), T()).SingleOrDefault
		End Function
End Module

File: Src/Icm.ContextConsole/TypeExtensions.cs
using System;
using System.Linq;
using System.Runtime.CompilerServices;

public static class TypeExtensions
{
	[Extension()]
	public static T[] GetAttributes<T>(Type type, bool inherit) where T : Attribute
	{
		return (T[])type.GetCustomAttributes(typeof(T), inherit);
	}
	[Extension()]
	public static bool HasAttribute<T>(Type type, bool inherit) where T : Attribute
	{
		return ((T[])type.GetCustomAttributes(typeof(T), inherit)).Count() > 0;
	}
	[Extension()]
	public static T GetAttribute<T>(Type type, bool inherit) where T : Attribute
	{
		return ((T[])type.GetCustomAttributes(typeof(T), inherit)).SingleOrDefault();
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
